2 Niagara Fundamentals Exercise - Multiplexer/multiplexer-rt/src/com/tridiumuniversity/multiplexer/BMultiplex.java

```diff
@@ -0,0 +1,251 @@
+package com.tridiumuniversity.multiplexer;
+
+import javax.baja.nre.annotations.NiagaraProperty;
+import javax.baja.nre.annotations.NiagaraType;
+import javax.baja.sys.*;
+
+@NiagaraType
+@NiagaraProperty(
+        name = "in0",
+        type = "BBoolean",
+        defaultValue = "BBoolean.FALSE",
+        flags = Flags.SUMMARY
+)
+@NiagaraProperty(
+        name = "in1",
+        type = "BBoolean",
+        defaultValue = "BBoolean.FALSE",
+        flags = Flags.SUMMARY
+)
+@NiagaraProperty(
+        name = "in2",
+        type = "BBoolean",
+        defaultValue = "BBoolean.FALSE",
+        flags = Flags.SUMMARY
+)
+@NiagaraProperty(
+        name = "in3",
+        type = "BBoolean",
+        defaultValue = "BBoolean.FALSE",
+        flags = Flags.SUMMARY
+)
+@NiagaraProperty(
+        name = "s0",
+        type = "BBoolean",
+        defaultValue = "BBoolean.FALSE",
+        flags = Flags.SUMMARY
+)
+@NiagaraProperty(
+        name = "s1",
+        type = "BBoolean",
+        defaultValue = "BBoolean.FALSE",
+        flags = Flags.SUMMARY
+)
+@NiagaraProperty(
+        name = "out",
+        type = "BBoolean",
+        defaultValue = "BBoolean.FALSE",
+        flags = Flags.READONLY | Flags.SUMMARY
+)
+
+public class BMultiplex extends BComponent{
+
+  public void changed(Property p, Context cx) {
+    if(!isRunning()) {
+      return;
+    }
+    if(!out.equals(p)) {
```

Comment:
Maybe this is personal preference, but I think it'd be better to check each individual Property you want to call `calculateOut()` for rather than check if the Property is not `out`. If another developer came along and added an `enabled` Property for instance to this component, they might not realize that now the output is going to get recalculated every time the user changes the enabled Property.